pkg/utils/handle_errors.go

```
@@ -45,6 +42,23 @@ var sqlError *mysql.MySQLError
 
 func HandleError(logger framework.Logger, c *gin.Context, err error) {
 	logger.Error(err)
+
+	// will not captured by sentry if its an explicit APIError
+	if apiErr, ok := err.(*api_errors.APIError); ok {
+		c.JSON(apiErr.StatusCode, gin.H{
+			"error": apiErr.Message,
+		})
+		return
+	}
+
+	// will not captured by sentry if its an record not found error from gorm
+	if errors.Is(err, gorm.ErrRecordNotFound) {
+		c.JSON(http.StatusNotFound, gin.H{
+			"error": err.Error(),
+		})
+		return
+	}
+
 	c.JSON(http.StatusInternalServerError, gin.H{
 		"error": err.Error(),
 	})
```

@rodanr 
Need to send generic error message in case the error is not handled properly. 
Sending the same error message as returned from db or other services has security risks.

---

Thank you @dipeshdulal sir for you suggestion. Suggested changes have been made. Thank you